### ISSUE TITLE
What's the meaning of accum_batch_size in ssd_pascal.py?

### ISSUE THREAD

**Issue by Lavender105** [weiliu89/caffe#598]
#ssd_pascal.py
batch_size = 2 # 32
accum_batch_size = 32
iter_size = accum_batch_size / batch_size

I don't understand the meaning of accum_batch_size here, and why it is equivalent to 32? If I train with my own dataset, should I modify the value of it?

**Comment by ujsyehao** [comment]
you know the importance of batch_size, If batch_size is too small, it can affect the step of gradient descent, If batch_size is too big, it has the error "out of cuda memory". So accum_batch_size can solve the problem.  Even if you set batch_size is small, we can use iter_size * batch_size.

**Comment by Lavender105** [comment]
could you please explain it further? I don't get this sentence "Even if you set batch_size is small, we can use iter_size * batch_size.".

**Comment by albertyou2** [comment]
same question here

**Comment by FLsur** [comment]
+1

**Comment by DL-85** [comment]
I also want to know the exact meaning of accum_batch_size.

**Comment by weiliu89** [comment]
It means it will only do the gradient update after processing accum_batch_size samples. You could check the code [here](https://github.com/weiliu89/caffe/blob/ssd/src/caffe/solver.cpp#L234-L239).

**Comment by ryusaeba** [comment]
Yes, it will accumulate gradient iter_size times, then do back propagation once.
Does anyone know will iter_size affect batchnorm layer?  I have this question is because batchnorm layer will update mean and variance during training but I am not sure will it update once after iter_size time?

**Comment by weiliu89** [comment]
I think it will. During forward time at training, it will only use current batch to compute mean and variance.

**Comment by jianbohuang** [comment]
It will compute mean,variance and moving_average_fraction at each time of iter_size. [BN layer](https://github.com/weiliu89/caffe/blob/4817bf8b4200b35ada8ed0dc378dceaf38c539e4/src/caffe/layers/batch_norm_layer.cpp#L107)·
If the BN is used,  the mini batch size in data layer has better larger than 16.
